x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/sort_stringify_json.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const sortObject = (jsObject) => {
+  if (typeof jsObject !== 'object' || jsObject === null) {
+    return jsObject;
+  }
+
+  if (Array.isArray(jsObject)) {
+    return jsObject.map((item) => sortObject(item));
+  }
+
+  return Object.keys(jsObject)
+    .sort()
+    .reduce((sorted, key) => {
+      sorted[key] = sortObject(jsObject[key]);
+      return sorted;
+    }, {});
+};
+
+export const sortAndStringifyJson = (jsObject: Record<string, unknown>): string =>
+  JSON.stringify(sortObject(jsObject), null, 2);
```

Comment:
@nikitaindik Thanks for fixing https://github.com/elastic/kibana/pull/171750#issuecomment-1831040485 using this code.

I think we can remove this complexity by using https://www.npmjs.com/package/json-stable-stringify. It's a popular library and it's already used in Kibana.

This code works (just tested it with all the libraries in your branch):

```ts
import stringify from 'json-stable-stringify';

export const sortAndStringifyJson = (jsObject: Record<string, unknown>): string =>
  stringify(jsObject, { space: 2 });
```